sql: create sqlbase package

RaduBerinde opened the pull request:
Moving `structured.proto`, `privilege.proto` to `sqlbase`, along with associated code and dependencies.

This will allow us to move `dist_sql` stuff to a separate package.

All changes are fairly mechanical, no code is significantly rewritten.

Recommend not using Reviewable - there are a lot of commits, and most of them are pretty small.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6530)

<!-- Reviewable:end -->

andreimatei commented:
so the idea is we'll have a new `dist_sql` package, and `sql` will depend on `dist_sql`? If so, does `dist_sql` need to dependend on privileges?

---

Review status: 0 of 73 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6530#-:-KH14ewgbLl9_A4BPRxE:1971775987)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Right. You mean on `sql/privilege`? I'm not sure, probably not, there's not much in there at the moment..

---

Review status: 0 of 73 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6530#-:-KH15L8ypPlW6R3L8Hmg:1987032649)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
LGTM
split today, split every day

RaduBerinde committed:
export descriptorProto and descriptorKey

RaduBerinde committed:
sql: move sql proto defs to sqlbase

Only refactoring the protos and associated code in this change.

RaduBerinde committed:
move formatversion_string.go

RaduBerinde committed:
sql: move code to sqlbase

Moving the go files which were associated with the protos in `sqlbase`, along
with the respective `.go` file and various pieces of code that they were
dependent on. The sqlbase package builds and the sqlbase tests work.

RaduBerinde committed:
adding sqlbase. as needed to sql files (1 of 2)

RaduBerinde committed:
export toEncodingDirection

RaduBerinde committed:
adding some missing comments for new exports

RaduBerinde committed:
adding sqlbase. in tests

RaduBerinde committed:
export descriptorTable

RaduBerinde committed:
export isSystemConfigID

RaduBerinde committed:
export wrapDescriptor

RaduBerinde committed:
export equalName

RaduBerinde committed:
export invalidMutationID

RaduBerinde committed:
export toDatumType, sanitizeDefaultExpr

RaduBerinde committed:
exporting errors

RaduBerinde committed:
export makeColumnDefDescs

RaduBerinde committed:
update uses of descriptorProto/Key

RaduBerinde committed:
change sql. to sqlbase. in other packages

RaduBerinde committed:
export stripColumnIDLength

RaduBerinde committed:
export methods on index, table descriptors

RaduBerinde committed:
export makeTableDesc

RaduBerinde committed:
export systemDB

RaduBerinde committed:
export allColumnsSelector

RaduBerinde committed:
adding sqlbase. as needed in sql files (2 of 2)

RaduBerinde committed:
fix sql. in other tests
